### PR TITLE
build: update Storybook to v7.3.0

### PR DESCRIPTION
**Pull request by henriettemoe**
Based on new Storybook changes.
Avoid deprecated ComponentMeta and ComponentStory.

### CODE REVIEW

File: h5p-bildetema/src/components/LanguageDropdown/LanguageDropdown.stories.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Meta, StoryFn } from "@storybook/react";
 import { LanguageDropdown, LanguageDropdownProps } from "./LanguageDropdown";
 
 export default {
   title: "Components/LanguageDropdown",
```
Comment (boyum) [inline h5p-bildetema/src/components/LanguageDropdown/LanguageDropdown.stories.tsx:5]:
```suggestion
  title: "Components/Language dropdown",
```

File: h5p-bildetema/src/components/LanguageSelector/LanguageSelector.stories.tsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Meta, StoryFn } from "@storybook/react";
 import { Language } from "common/types/types";
 import { LanguageSelector, LanguageSelectorProps } from "./LanguageSelector";
 
 export default {
   title: "Components/LanguageSelector",
```
Comment (boyum) [inline h5p-bildetema/src/components/LanguageSelector/LanguageSelector.stories.tsx:6]:
```suggestion
  title: "Components/Language selector",
```

File: h5p-bildetema/src/components/LanguageSelectorElement/LanguageSelectorElement.stories.tsx [hunk 3]
```diff
@@ -1,12 +1,12 @@
-import { ComponentMeta } from "@storybook/react";
+import { Meta } from "@storybook/react";
 import { LanguageCode } from "common/types/LanguageCode";
 import { Language } from "common/types/types";
 import { LanguageSelectorElement } from "./LanguageSelectorElement";
 
 export default {
   title: "Components/LanguageSelectorElement",
```
Comment (boyum) [inline h5p-bildetema/src/components/LanguageSelectorElement/LanguageSelectorElement.stories.tsx:7]:
```suggestion
  title: "Components/Language selector element",
```

File: h5p-bildetema/src/components/PrintButton/PrintButton.stories.tsx [hunk 4]
```diff
@@ -1,10 +1,10 @@
-import { ComponentMeta } from "@storybook/react";
+import { Meta } from "@storybook/react";
 import { PrintButton } from "./PrintButton";
 
 export default {
   title: "Components/PrintButton",
```
Comment (boyum) [inline h5p-bildetema/src/components/PrintButton/PrintButton.stories.tsx:5]:
```suggestion
  title: "Components/Print button",
```

File: h5p-bildetema/src/components/SubHeader/SubHeader.stories.tsx [hunk 5]
```diff
@@ -1,12 +1,12 @@
-import { ComponentMeta } from "@storybook/react";
+import { Meta } from "@storybook/react";
 import { TopicGridSizes } from "common/types/types";
 import { useState } from "react";
 import { SubHeader } from "./SubHeader";
 
 export default {
   title: "Components/SubHeader",
```
Comment (boyum) [inline h5p-bildetema/src/components/SubHeader/SubHeader.stories.tsx:7]:
```suggestion
  title: "Components/Sub header",
```

File: h5p-bildetema/src/components/TopicGrid/TopicGrid.stories.tsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
-import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Meta, StoryFn } from "@storybook/react";
 import { LanguageCode } from "common/types/LanguageCode";
 import { Topic, TopicGridSizes, Word } from "common/types/types";
 import { TopicGrid, TopicGridProps } from "./TopicGrid";
 
 export default {
-  label: "Molecules/ToppicGrid",
+  title: "Molecules/TopicGrid",
```
Comment (boyum) [inline h5p-bildetema/src/components/TopicGrid/TopicGrid.stories.tsx:7]:
```suggestion
  title: "Molecules/Topic grid",
```

File: h5p-editor-bildetema-words-topic-image/src/components/ColorButton/ColorButton.stories.tsx [hunk 7]
```diff
@@ -1,13 +1,13 @@
-import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Meta, StoryFn } from "@storybook/react";
 import { Color } from "common/enums/Color";
 import { ColorButton, ColorButtonProps } from "./ColorButton";
 
 export default {
   title: "Components/ColorButton",
```
Comment (boyum) [inline h5p-editor-bildetema-words-topic-image/src/components/ColorButton/ColorButton.stories.tsx:6]:
```suggestion
  title: "Components/Color button",
```

File: h5p-editor-bildetema-words-topic-image/src/components/TopicChooser/TopicChooser.stories.tsx [hunk 8]
```diff
@@ -1,12 +1,12 @@
-import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Meta, StoryFn } from "@storybook/react";
 import { LanguageCode } from "common/types/LanguageCode";
 import { Topic, Word } from "common/types/types";
 import { TopicChooser } from "./TopicChooser";
 
 export default {
-  label: "Molecules/TopicChooser",
+  title: "Molecules/TopicChooser",
```
Comment (boyum) [inline h5p-editor-bildetema-words-topic-image/src/components/TopicChooser/TopicChooser.stories.tsx:7]:
```suggestion
  title: "Molecules/Topic chooser",
```

**Review by boyum — APPROVED** [review]
Nice!!!

**Comment by boyum** [comment]
Instead of binding templates we could also export each story directly by moving the template to the default export's `render` property:

Before:

```tsx
export default {
  title: "Component",
  component: Component,
} satisfies Meta<typeof Component>

const Template: StoryFn<typeof Component> = args => <Component {...args} />;

export const MyStory1 = Template.bind({});
MyStory.args = {
  hello: "world",
};

export const MyStory2 = Template.bind({});
MyStory.args = {
  hello: "world",
};
```

After:

```tsx
export default {
  title: "Component",
  component: Component,
  render: args => <Component {...args} />,
} satisfies Meta<typeof Component>

type Story = StoryFn<typeof Component>;

export const MyStory1: Story = {
  args: {
    hello: "world",
  }
};

export const MyStory2: Story = {
  args: {
    hello: "world",
  }
};
```

**Comment by boyum** [comment]
It's perhaps possible to run `npx storybook@latest migrate csf-2-to-3 --glob="**/*.stories.tsx"` to automatically migrate all stories to the latest format. https://storybook.js.org/blog/storybook-csf3-is-here/

**Comment by henriettemoe** [comment]
> It's perhaps possible to run `npx storybook@latest migrate csf-2-to-3 --glob="**/*.stories.tsx"` to automatically migrate all stories to the latest format. https://storybook.js.org/blog/storybook-csf3-is-here/

Yes, seems to work. But it also says "Need to install the following packages: storybook@7.3.0" to do so, even though the command itself does not update the package

**Review by boyum — APPROVED** [review]

**Review by boyum — APPROVED** [review]

**Pull request merged**